ts/src/bingx.ts

```diff
@@ -0,0 +1,2721 @@
+        //        "data": {
+        //            "longLeverage": 6,
+        //            "shortLeverage": 6
+        //        }
+        //    }
+        //
+        return response;
+    }
+
+    async setLeverage (leverage, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#setLeverage
+         * @description set the level of leverage for a market
+         * @see https://bingx-api.github.io/docs/#/swapV2/trade-api.html#Switch%20Leverage
+         * @param {float} leverage the rate of leverage
+         * @param {string} symbol unified market symbol
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} response from the exchange
+         */
+        this.checkRequiredSymbol ('setLeverage', symbol);
+        const side = this.safeStringUpper (params, 'side');
+        this.checkRequiredArgument ('setLeverage', side, 'side', [ 'LONG', 'SHORT' ]);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+            'side': side,
+            'leverage': leverage,
+        };
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": {
+        //            "leverage": 6,
+        //            "symbol": "BTC-USDT"
+        //        }
+        //    }
+        //
+        return await this.swapV2PrivatePostTradeLeverage (this.extend (request, params));
+    }
+
+    async fetchMyTrades (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#fetchMyTrades
+         * @description fetch all trades made by the user
+         * @see https://bingx-api.github.io/docs/#/swapV2/trade-api.html#Query%20historical%20transaction%20orders
+         * @param {string|undefined} symbol unified market symbol
+         * @param {int|undefined} since the earliest time in ms to fetch trades for
+         * @param {int|undefined} limit the maximum number of trades structures to retrieve
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @param {string} params.trandingUnit COIN (directly represent assets such as BTC and ETH) or CONT (represents the number of contract sheets)
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=trade-structure}
+         */
+        this.checkRequiredArgument ('fetchMyTrades', symbol, 'symbol');
+        this.checkRequiredArgument ('fetchMyTrades', since, 'since');
+        const tradingUnit = this.safeStringUpper (params, 'tradingUnit');
+        if (tradingUnit === undefined) {
+            throw new BadRequest (this.id + ' fetchMyTrades() requires either COIN (directly represent assets such as BTC and ETH) or CONT (represents the number of contract sheets) for tradingUnit');
```

Comment:
@AlgoQ we can assume a default value here, we don't need to block the call because of this.  For swap markets we can set `cont` as the default